frontend/src/routes/events/EventPage.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from 'react'
+import { useGeneralContext } from '../../contexts/generalState'
+import { EventType } from '../../data/event'
+import ScrollToTop from '../../components/ScrollToTop/ScrollToTop'
+import { useParams, useNavigate, useLocation } from 'react-router-dom'
+import { getEvent } from '../../api/event'
+import { getFormattedPeriod } from '../../util/functions'
+import BackButton from '../../components/BackButton/BackButton'
+
+import './EventPage.scss'
+
+interface EventState {
+  event: EventType | null
+  loading: boolean
+}
+interface CustomState {
+  from: string
+}
+
+function EventPage() {
+  const [eventRequest, setEventRequest] = useState<EventState>({ event: null, loading: true })
+  const { backgroundImages } = useGeneralContext()
+  const { eventId } = useParams()
+  const navigate = useNavigate()
+  const location = useLocation()
+
+  const backBtnText = location.state && (location.state as CustomState).from === 'home' ? 'Acasă' : 'Evenimente'
+
+  useEffect(() => {
+    ;(async () => {
```

Comment:
Why is an IIFE needed here?

Reply:
because I need an asynchronous function, but I changed it to use then everywhere instead of await